addons/mail/models/discuss/mail_thread.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from markupsafe import Markup
+
+from odoo import models, _
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    def _notify_thread(self, message, msg_vals=False, **kwargs):
+        rdata = super()._notify_thread(message, msg_vals, **kwargs)
+        self._notify_mentioned_channel(**kwargs)
+        return rdata
+
+    def _notify_mentioned_channel(self, channel_ids=None, **kwargs):
+        """ Notify channels that have been mentioned in message"""
+        if channel_ids:
+            body = Markup(_("%(author)s mentioned this channel in %(record)s")) % {
+                "author": self.env.user.partner_id._get_html_link(),
+                "record": Markup('<a href="#" data-oe-type="notification" data-oe-id="%s" data-oe-model="%s">%s</a>') % (
+                    self.id, self._name, self.display_name
+                )
+            }
+            for channel in self.env['discuss.channel'].browse(channel_ids):
```

Comment:
I would just keep "This channel was mentioned in X":
- The user name being a link serves no purpose, clicking does nothing
- Using `partner.display_name` is too verbose compared to `partner.name`. This is the difference between "YourCompany, Marc Demo" vs "Marc Demo".
- The shorter the notification the better, the most important aspect is to tell channel has been mentioned and then where. Author is not imported and can be easily found by clicking on link to chatter.